### ISSUE TITLE
Child Steps error

### ISSUE THREAD

**Issue by chathsuom** [OctopusDeployLabs/terraform-provider-octopusdeploy#261]
**Describe the bug**
Trying to deploy a step with multiple actions (actual requirement is to have child steps)

**Steps to reproduce**
```resource "octopusdeploy_deployment_process" "this" {
  project_id = module.ecs_projects["gayan_test"].id
  step {
    condition           = "Success"
    name                = "Hello world (using PowerShell)"
    package_requirement = "LetOctopusDecide"
    start_trigger       = "StartAfterPrevious"
    action {
        action_type  = "Octopus.Script"
      can_be_used_for_project_versioning = false
      condition                          = "Success"
      is_disabled                        = false
      is_required                        = true
      name                               = "Hello world (using PowerShell)"
      run_on_server                      = true
      properties = {
          "Octopus.Action.Script.ScriptBody"= "echo hello"
      }
      worker_pool_id = "WorkerPools-22"
      action_template {
                id      = "ActionTemplates-62"
                version = 41
        }
    }
    action {
        action_type  = "Octopus.Script"
      can_be_used_for_project_versioning = false
      condition                          = "Success"
      is_disabled                        = false
      is_required                        = true
      name                               = "test2"
      run_on_server                      = true
      properties = {
          "Octopus.Action.Script.ScriptBody"= "echo hello"
      }
      worker_pool_id = "WorkerPools-22"
      action_template {
                id      = "ActionTemplates-106"
                version = 4
        }
    }
}

**Expected behavior**
Succesful execution

**Logs and other supporting information**
```│ Error: octopus deploy api returned an error on endpoint /api/Spaces-1/deploymentprocesses/deploymentprocess-Projects-176 - [Please select one or more roles that 'Hello world (using PowerShell)' step will apply to.]
│ 
│   with octopusdeploy_deployment_process.this,
│   on ecs_projects.tf line 405, in resource "octopusdeploy_deployment_process" "this":
│  405: resource "octopusdeploy_deployment_process" "this" {
```

**Screenshots**

**Environment and versions:**
```
Model Name:	MacBook Pro
  Model Identifier:	MacBookPro17,1
  Chip:	Apple M1
  Total Number of Cores:	8 (4 performance and 4 efficiency)
  Memory:	16 GB
  System Firmware Version:	6723.140.2
  OS Loader Version:	6723.140.2
  Terraform v1.0.8
  on darwin_arm64
   provider registry.terraform.io/octopusdeploylabs/octopusdeploy v0.7.58
```

**Comment by jbristowe** [comment]
Did adding a `roles` attribute/value pair resolve this issue?

**Comment by chathsuom** [comment]
With the latest version with worker pool enabled, roles issues disappeared. however, I still don't get how to add child steps.
<img width="396" alt="image" src="https://user-images.githubusercontent.com/1393675/150046567-43a83ea9-92b1-4afd-bf60-58d61c374ca8.png">

**Comment by chathsuom** [comment]
```      ~ step {
            id                   = "e61f7372-3672-4b28-b79a-4cf5f5a2996b"
            name                 = "Deploy AWS Resources"
            # (6 unchanged attributes hidden)

          ~ action {
                id                                 = "f9fb7244-a0c4-41ed-b2c9-b8fd94190215"
              ~ name                               = "Deploy AWS Resources3" -> "Deploy AWS Resources4"
                # (12 unchanged attributes hidden)


                # (2 unchanged blocks hidden)
            }
          + action {
              + action_type           = "Octopus.AwsRunCloudFormation"
              + channels              = (known after apply)
              + condition             = "Success"
              + environments          = (known after apply)
              + excluded_environments = [
                  + "Environments-31",
                ]```

Each run it always shows this for plan

**Comment by davidkarim** [comment]
I need the capability to add child steps as well because in my team, we are trying to implement rolling deploys. Is this possible to do today? If not, can this be added? We are in need of this capability today.

**Comment by hnrkndrssn** [comment]
Hi @chathsuom and @davidkarim I've looked into this and the following terraform config creates a deployment process with a step with child steps.

```
resource "octopusdeploy_deployment_process" "TheDeploymentProcess" {
  project_id = octopusdeploy_project.TheProject.id
  step {
    condition = "Success"
    name = "Run scripts"
    package_requirement = "LetOctopusDecide"
    start_trigger = "StartAfterPrevious"
    target_roles = [ "web" ]
    run_script_action {
      can_be_used_for_project_versioning = false
      condition = "Success"
      is_disabled = false
      is_required = true
      name = "Say Hello"
      script_syntax = "Bash"
      script_body = <<-EOT
        echo 'Hello'
      EOT
    }
    run_script_action {
      can_be_used_for_project_versioning = false
      condition = "Success"
      is_disabled = false
      is_required = true
      name = "Say Goodbye"
      script_syntax = "Bash"
      script_body = <<-EOT
        echo 'Goodbye'
      EOT
    }
  }
}
```

@davidkarim to make this a rolling deploy you have to add the `window_size` property on the `step` resource to how many machines you want to run in parallel.

I hope that helps!

**Comment by chathsuom** [comment]
Thanks. Actually, we get rid of using child steps for the moment. I will note this and you can close this one if that's suits.

**Comment by hnrkndrssn** [comment]
Thanks for letting us know @chathsuom. I will close this for now, if you run into any issue in the future please raise a new issue.

**Comment by davidkarim** [comment]
@hnrkndrssn Wow! Thank you! Just tested it and it looks like it works. I am not sure if I am missing something regarding the documentation, but I am thinking this wasn't documented anywhere. I think the docs need some love. But anyways, thanks for the help.
